### ISSUE TITLE
PR test in mdn/translated-content is slower than before

### ISSUE THREAD

**Issue by yin1999** [mdn/yari#8301]
### Summary

I just notice that after we migrating to ESM. The PR test workflow in translated-content is slower that before.

See the following workflow run:

- now: https://github.com/mdn/translated-content/actions/runs/4261747120/jobs/7416424977#step:7:63
- before: https://github.com/mdn/translated-content/actions/runs/4048314445/jobs/6963395796#step:7:30

As [@queengooborg mentioned in translated-content](https://github.com/mdn/translated-content/pull/11876#issuecomment-1443517259), maybe we need synchronous file reading for this step.

### URL

None

### Reproduction steps

See workflow run.

### Expected behavior

The workflow could run faster (maybe we could start from the gather translation step).

### Actual behavior

The workflow is a bit slow.

### Device

Desktop

### Browser

Chrome

### Browser version

Stable

### Operating system

Linux

### Validations

- [X] I have read the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
- [X] I have verified that there isn't [already an issue](https://github.com/mdn/yari/issues) that reports the same bug to avoid creating a duplicate.
- [X] I have checked that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/orgs/mdn/discussions/categories/platform).

**Comment by caugner** [comment]
Please note that individual workflow runs are not necessarily comparable, as workflow runners use a variety of CPUs.

**Comment by yin1999** [comment]
> Please note that individual workflow runs are not necessarily comparable, as workflow runners use a variety of CPUs.

Sorry about ignoring that, I just made a locale test, and found that:

use `@mdn/yari@2.3.0`:

`13.639s` is used to gather translations.

![image](https://user-images.githubusercontent.com/15844309/221179031-3a1cec5e-d7e6-4043-93f3-63913cd4cd6e.png)

use `@mdn/yari@1.36.2`:

`1.831s` is used to gather translations.

![image](https://user-images.githubusercontent.com/15844309/221178777-6cbe3207-3a10-49b2-bef2-245a503aa967.png)


Test platform:

```plain
OS: Linux debian 6.1.0-3-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.8-1 (2023-01-29) x86_64 GNU/Linux
Node.js: v18.13.0
Yarn: 1.22.19 (Debian package: yarnpkg@1.13.0-1+deb10u1)
CPU: 13th Gen Intel(R) Core(TM) i5-13600K

With 1TB SSD and 32GB memory
```

**Comment by caugner** [comment]
I compared `@mdn/yari@1.36.4` and `@mdn/yari@2.0.0` and indeed the build performance has degraded significantly:

- 1.36.4: 19.0 seconds
- 2.0.0: 49.0 seconds

**Comment by queengooborg** [comment]
On bf23810fffdd4ba49c38d60a4f7ec982106e2ff3, I ran `yarn build files/zh-tw/learn/common_questions/web_mechanics/what_is_a_web_server/index.md` and the command completed in ~19 seconds:

```
❯ yarn build files/zh-tw/learn/common_questions/web_mechanics/what_is_a_web_server/index.md
yarn run v1.22.19
$ cross-env NODE_ENV=production ts-node build/cli.ts files/zh-tw/learn/common_questions/web_mechanics/what_is_a_web_server/index.md
(node:36300) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
CONTENT_ROOT:            /Users/queengooborg/Developer/Gooborg/mdn-content/files
CONTENT_TRANSLATED_ROOT: /Users/queengooborg/Developer/Gooborg/mdn-translated-content/files
Time to gather all translations: 16.302s
Language does not exist: svelte
/Users/queengooborg/Developer/Gooborg/yari/client/build/zh-tw/docs/learn/common_questions/web_mechanics/what_is_a_web_server
Built 1 pages in 17.5 seconds, at a rate of 0.1 documents per second.
Peak heap memory usage: 300 MB


Total_Flaws_Count
broken_links  8
macros        1


✨  Done in 19.45s.
```

The time spent gathering all the translations is definitely quite high, I wonder what changed in between v1 and v2...

**Comment by queengooborg** [comment]
Now this is interesting (result from running a profiler through Chrome's debug panel):

<img width="1285" alt="Screenshot 2023-02-25 at 05 26 40" src="https://user-images.githubusercontent.com/5179191/221359392-878c2dd3-cc98-45ba-8129-5c87ab091d11.png">

**Comment by yin1999** [comment]
After the fix (#8806) is applied, the build time is reduced to `3.5s`. Let's see if there is side-effect made by the fix for a few days. If nothing happend, I'll close this PR :)

![image](https://github.com/mdn/yari/assets/15844309/4487b273-2372-4569-8dda-ea63c26b3543)
